### ISSUE TITLE
Universal wheels for Python 3.11 crash on macOS arm

### ISSUE THREAD

**Issue by PhilipVinc** [pyscf/pyscf#1791]
on Python 3.11, ARM processors on Mac pip, when installing psycf, will automatically pickup the universal wheels [that you uploaded to PyPi](https://pypi.org/project/pyscf/2.3.0/#files).

> pyscf-2.3.0-cp311-cp311-macosx_10_9_universal2.whl

This would be great, as it prevents users having from having to build psych themselves.
However, it seems that those wheels are incorrectly tagged as compatible with arm, as if I try to load pyscf in python I get the following DLOpen error 

```python
File ~/.pyenv/versions/3.11.2/lib/python3.11/ctypes/__init__.py:376, in CDLL.__init__(self, name, mode, handle, use_errno, use_last_error, winmode)
    373 self._FuncPtr = _FuncPtr
    375 if handle is None:
--> 376     self._handle = _dlopen(self._name, mode)
    377 else:
    378     self._handle = handle
```
> OSError: dlopen(/Users/filippo.vicentini/Documents/pythonenvs/netket/python-3.11.2/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib, 0x0006): tried: '/Users/filippo.vicentini/Documents/pythonenvs/netket/python-3.11.2/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/filippo.vicentini/Documents/pythonenvs/netket/python-3.11.2/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (no such file), '/Users/filippo.vicentini/Documents/pythonenvs/netket/python-3.11.2/lib/python3.11/site-packages/pyscf/lib/libnp_helper.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))

I think that at least you should yank that wheel so that pip does not install a non-working wheel on macOS arm.

**Comment by verena-neufeld** [comment]
Thanks @PhilipVinc for raising this issue. Looks like it has been resolved?

**Comment by PhilipVinc** [comment]
Assuming the wheels generated by that workflow are correct, yes.
All that is left would be to run that action once to upload new wheels. Maybe you could tag a prerelease version (v2.3.0.pre1) which will not get installed by users but can check that all is working correctly.

It would be beautiful to support all architectures through cibuildwheel but that's already a great step forward!

**Comment by mkrompiec** [comment]
It would be great if the updated wheels got uploaded soon, this is stopping us from moving our workflow to Python 3.11

**Comment by sunqm** [comment]
Wheels for Apple silicon have been available since Pyscf-2.5. If you encounter a similar issue in the new release, please feel free to reopen this issue.
